### ISSUE TITLE
Handling cascaded generics call

### ISSUE THREAD

**Issue by csyonghe** [shader-slang/slang#253]
lower-to-ir crashes when trying to lower a substitution argument that is a GenericConstraintDecl.

Test case:

```
//TEST(compute):COMPARE_COMPUTE:-xslang -use-ir
//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):dxbinding(0),glbinding(0),out

RWStructuredBuffer<float> outputBuffer;

interface IBase
{
    float get();
}

struct BaseImpl : IBase
{
    float get() { return 1.0; }
};

__generic<T:IBase>
float eval(T obj)
{
    return obj.get();
}

__generic<T : IBase>
float test(T obj)
{
    return eval<T>(obj);
}

[numthreads(4, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
    BaseImpl base;
	float outVal = test<BaseImpl>(base);
	outputBuffer[dispatchThreadID.x] = outVal;
}
```

Failed at lowering the `eval<T>(obj)` call when lowering `eval<T>` when lowering the substitution arg `T:IBase`.

**Comment by tfoleyNV** [comment]
The argument isn't just going to be a `DeclRef`, because it needs to be a `Val`, right? Ideally we should be seeing the argument be a `SubtypeWitness`, and we should handle it just like we handle the case for the non-"cascaded" generic call `test<BaseImpl>`.

I wonder what is going wrong in this case...

**Comment by csyonghe** [comment]
It is crashing because there is no visitGenericConstraintDecl() in DeclLoweringVisitor, causing visitDecl() being called when the code attempts to lower the substitution argument into a IRValue. The visitDecl() function just throws InternalError.

**Comment by csyonghe** [comment]
I am not sure what to do in this case. What should a genericConstraintDecl translate into?

**Comment by csyonghe** [comment]
This issue is fixed in PR #251
